pkg/ebpf/c/tracee.bpf.c

```diff
@@ -5391,6 +5392,32 @@ int BPF_KPROBE(trace_security_file_permission)
     return 0;
 }
 
+SEC("raw_tracepoint/task_rename")
+int tracepoint__task__task_rename(struct bpf_raw_tracepoint_args *ctx)
+{
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+        return 0;
+    if (!should_trace((&data.context)))
+        return 0;
+
+    struct task_struct *tsk = (struct task_struct *)ctx->args[0];
+    char old_name[TASK_COMM_LEN];
+    bpf_probe_read_str(&old_name, TASK_COMM_LEN, tsk->comm);
+    const char *new_name = (const char *)ctx->args[1];
+
+    // this tracepoint is invoked for task renaming - this also happens when a process first starts.
+    // so check if this is new execution or not.
+    // this arg was inspired by __set_task_comm() - https://elixir.bootlin.com/linux/v5.18.2/source/fs/exec.c#L1228.
+    bool exec = (get_task_host_pid(tsk) == get_task_host_tgid(tsk));
```

Comment:
Is that always true?
What if the main thread (which has tgid) renamed itself?

Reply:
your'e right, good point.

i think the way to tackle this is to create a map indicating an exec. this map would be updated in `sched_process_fork`, and deleted from in `sched_process_exec` and `task_rename`.

another is option is to go back to using kprobe `__set_task_comm` (https://elixir.bootlin.com/linux/latest/source/fs/exec.c#L1228) which get this information as an argument.

WDYT?

Reply:
can't we just check if the current syscall is execve/execveat?

Reply:
yes we can